Lib/urllib/parse.py

```diff
@@ -860,8 +860,7 @@ def splithost(url):
     """splithost('//host[:port]/path') --> 'host[:port]', '/path'."""
     global _hostprog
     if _hostprog is None:
-        import re
-        _hostprog = re.compile('^//([^/?]*)(.*)$')
+        _hostprog = re.compile('//([^/#?]*)(.*)', re.DOTALL)
```

Comment:
I'm not sure sure of the addition of DOTALL, see:
http://bugs.python.org/issue30500#msg296422